kerckhoff/users/serializers.py

```diff
@@ -1,17 +1,50 @@
 from rest_framework import serializers
 from .models import User
+from kerckhoff.userprofiles.serializers import UserProfileSerializer
+from kerckhoff.userprofiles.models import UserProfile
 
 
 class UserSerializer(serializers.ModelSerializer):
+    userprofile = UserProfileSerializer()
 
     class Meta:
         model = User
-        fields = ('id', 'username', 'first_name', 'last_name',)
-        read_only_fields = ('username', )
+        fields = ("id", "username", "first_name", "last_name", "userprofile")
+        read_only_fields = ("username",)
 
+    def create(self, validated_data):
+        userprofile_data = validated_data.pop("userprofile")
+        user = User.objects.create(**validated_data)
+        UserProfile.objects.create(user=user, **userprofile_data)
+        return user
 
-class CreateUserSerializer(serializers.ModelSerializer):
+    def update(self, instance, validated_data):
+        userprofile_data = validated_data.pop("userprofile")
+        userprofile = instance.userprofile
+
+        instance.first_name = validated_data.get("first_name", instance.first_name)
+        instance.last_name = validated_data.get("last_name", instance.last_name)
+        instance.save()
 
+        userprofile.title = userprofile_data.get("title", userprofile.title)
```

Comment:
same, but userprofile.update(). (btw I cant exactly remember if you can do this (for this and the above), the alternative is to do `UserProfile.objects.filter(pk=userprofile.id).update(**userprofile_data)`